pkgs/applications/networking/mailreaders/meli/default.nix

```diff
@@ -39,6 +40,8 @@ rustPlatform.buildRustPackage rec {
     mkdir -p $out/share/man/man5
     gzip < docs/meli.conf.5 > $out/share/man/man5/meli.conf.5.gz
     gzip < docs/meli-themes.5 > $out/share/man/man5/meli-themes.5.gz
+  '' + lib.optionalString withNotmuch ''
+    wrapProgram $out/bin/meli --set LD_LIBRARY_PATH ${notmuch}/lib
```

Comment:
How come it doesn't link to it?

Reply:
Yeah I also wondered about it. For some reason they load it at runtime in this way: https://git.meli.delivery/meli/meli/src/branch/master/melib/src/backends.rs#L159

Probably so they can distribute meli without building against libnotmuch?

Reply:
Yeah but if it doesn't even start without it loading it won't do anything right? Maybe it's a rust thing, where they don't have a crate that wraps the library.. IDK

Reply:
I'd put a comment above the `wrapProgram` command, otherwise LGTM.

Reply:
Mh what do you mean with "won't do anything"? libnotmuch is just one of many backends. At startup it reads my config file where I configured to use the notmuch backend.
Tbh I don't know much about how Rust loads non-Rust libraries. I just saw that wrapper pattern quite frequently in Nixpkgs and derived from it that it may be our problem, not upstream's.

Reply:
Alright, added a comment.

Reply:
Oh I see, so not necessarily you have notmuch configured in your `~/.melirc` (or whatever) right?

Reply:
Yeah exactly, you could configure other ways to access mail in `~/.config/meli/config.toml` and I guess they work fine. I never tested them though.